Core/src/main/java/com/paypal/android/core/HttpRequestFactory.kt

```diff
@@ -0,0 +1,26 @@
+package com.paypal.android.core
+
+import java.net.URL
+
+class HttpRequestFactory {
+
+    fun createHttpRequestFromAPIRequest(
+        apiRequest: APIRequest,
+        configuration: PaymentsConfiguration
+    ): HttpRequest {
+        val path = apiRequest.path
+        val baseUrl = configuration.environment.url
+
+        val url = URL("$baseUrl/$path")
+        val method = apiRequest.method
+        val body = apiRequest.body
+
+        val httpRequest = HttpRequest(url, method, body)
+        if (method == HttpMethod.POST) {
+            httpRequest.contentType = HttpContentType.JSON
+        }
+
+        httpRequest.headers["Authorization"] = "Bearer ${configuration.authToken}"
```

Comment:
Aren't we going to use `"Basic <clientId>:" for all of our network requests?

Reply:
Oh we never tried with a trailing `:` does this currently work for the `confirm-payment-source` route?

Reply:
`confirm-payment-source` currently requires an upgraded access token - on iOS we added a 'secret' field to be populated while activelytesting: https://github.com/paypal/iOS-SDK/pull/13/files#diff-c0a4c09ae6d8bbf7cdf018e06a195f26c3fd22604cc61e385cd2d1ab1400d04fR19-R20

Reply:
Ah yeah I remember now. We could go for something similar to keep consistent with iOS. Is the plan to only support client ID auth in our first release, or would the merchant be using access tokens?

Reply:
Plan is support client ID basic auth on release, pending the API team lifting the requirement for the `confirm-payment-source` endpoint having an upgraded token.

Reply:
Cool yeah we migrated to Basic Auth with `base64(CLIENT_ID:CLIENT_SECRET)`.